.github/workflows/build-and-run-gpu.yml

```
@@ -0,0 +1,124 @@
+name: Build-And-Run Tests on GPU
+
+on:
+  pull_request:
+    # Trigger the workflow whenever a PR receives a special label
+    # in addition to the default triggering events.
+    types: [ labeled, opened, synchronize, reopened ]
+
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref }}
+  cancel-in-progress: true
+
+jobs:
+  build_and_test_gpu:
+    # Trigger on PRs with specific label
+    if: contains(github.event.pull_request.labels.*.name, 'GPU')
+
+    env:
+      ALL_EXCLUDE: stage_dem|pybind\.unwrap\.phass
+      # stage_dem requires AWS permissions to access the S3 bucket containing the DEM data.
+      # pybind.unwrap.phass occasionally segfaults on GHA runners.
+
+    defaults:
+      run:
+        shell: bash -l {0}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - label: Linux
+            runner: ubuntu-gpu-t4
+            ctest_exclude: -E ".*($ALL_EXCLUDE)"
+        deps:
+          - label: Latest
+            env-file: environment.yml
+            extra-specs: ""
+        build-type: [ RelWithDebInfo ]
```

I don't recall when we decided which build-types we'd use in the GHA CI system, but the GPU runner uses RelWithDebInfo while the CPU runners use either Release or Debug - could you document the reasoning for these selections in the files?

---

Sure - Release and Debug are two opposite extremes so it's nice to test with both when possible, so that's what I'm doing for the free CPU-only builds. The GPU builds cost us per-minute, so I'm setting up a single GPU build and avoiding a whole matrix of possibilities for the build. RelWithDebInfo makes the tests fast but still builds with debugging checks, so it seems like the best compromise if you need to just pick one.

---

Good - let's add a comment here and in the same line on CPU that explains this decision since it seems like it could be useful to refresh on when you or someone else come back to modify the file later on
```suggestion
        # It would be ideal to test with both Release and Debug build types, but GPU runners cost per-minute;
        # to avoid excess runtime costs, we compromise to run with RelWithDebInfo build type, which runs fast
        # but still produces debugging checks.
        build-type: [ RelWithDebInfo ]
```